src/libgap-api.c

```diff
@@ -0,0 +1,127 @@
+// LibGAP API - API for using GAP as shared library.
+
+#include "libgap-api.h"
+
+#include "bool.h"
+#include "calls.h"
+#include "gapstate.h"
+#include "gvars.h"
+#include "intobj.h"
+#include "opers.h"
+#include "plist.h"
+#include "stringobj.h"
+
+Obj FuncREAD_ALL_COMMANDS(Obj self, Obj instream, Obj echo, Obj capture, Obj resultCallback);
+
+// Setup and initialisation
+
+void GAP_Initialize(int          argc,
+                    char **      argv,
+                    char **      env,
+                    CallbackFunc markBagsCallback,
+                    CallbackFunc errorCallback)
+{
+    InitializeGap(&argc, argv, env);
```

Comment:
That would be my preferred goal. Figuring out what the correct API is is the current challenge.

Reply:
Exactly.

There are two things to concern: How stable do we want the API to be? Having separate function allows us to have a different version for the API than for GAP itself. This is imho a pro for separate functions.

I think using the kernel functions directly would really force us to prefix them all. I am not against this, but it might break a lot of code.